pkg/operator/controller/controller.go

```diff
@@ -186,6 +186,12 @@ func (r *ReconcileCDI) Reconcile(request reconcile.Request) (reconcile.Result, e
 		return reconcile.Result{}, err
 	}
 
+	if configMap != nil && configMap.DeletionTimestamp != nil {
+		// this is a left over from a previous CR, wait for it get deleted.
+		reqLogger.Info("Found previous config that is marked for deletion, requeueing")
+		return reconcile.Result{RequeueAfter: time.Second}, nil
```

Comment:
why a Second?

Reply:
Basically to prevent us from looping really fast until the configmap is actually gone. I suspect this might occur on a highly loaded control plane. So instead of adding to the load, just wait a little bit.